Pomodoro.Services/StatisticsService.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Pomodoro.Core.Enums;
+using Pomodoro.Core.Interfaces.IServices;
+using Pomodoro.Core.Models.Statistics;
+using Pomodoro.DataAccess.EF;
+using Pomodoro.DataAccess.Entities;
+
+namespace Pomodoro.Services
+{
+	public  class StatisticsService : IStatisticsService
+	{
+		public async Task<DailyStatistics> GetDailyStatisticsAsync(Guid userId, DateOnly day)
+		{
+			DailyStatistics dailyStatistics = new DailyStatistics();
+
+			using (AppDbContext db = new AppDbContext())
+			{
+				IQueryable<Completed> tasks = db.CompletedTasks;
+				
+				tasks = tasks.Where(t => t.Id == userId);
+				tasks = tasks.Where(t => t.ActualDate.Year == day.Year);
+				tasks = tasks.Where(t => t.ActualDate.Month == day.Month);
+				tasks = tasks.Where(t => t.ActualDate.Day == day.Day);
+
+
+				for (int i = 0; i < 24; i++)
+				{
+					int hour = i;
+					
+					AnalyticsPerHour analytics = new AnalyticsPerHour();
+
+					IQueryable<Completed> hourstatistic = tasks.Where(t => t.ActualDate.Hour == hour);
+
+					analytics.Hour = hour;
+
+					foreach (var h in hourstatistic)
+					{
+						analytics.TimeSpent += h.TimeSpent;
+						analytics.PomodorosDone += h.PomodorosCount;
+					}
+
+					dailyStatistics.AnalyticsPerHours?.Add(analytics);
+				}
+			}
+			return await Task.FromResult(dailyStatistics);
+
+		}
+
+		public async Task<MonthlyStatistics> GetMonthlyStatisticsAsync(Guid userId, int year, int month)
+		{
+			MonthlyStatistics monthlyStatistics = new MonthlyStatistics();
+
+			monthlyStatistics.UserId = userId;
+
+			using (AppDbContext db = new AppDbContext())
+			{
+				IQueryable<Completed> tasks = db.CompletedTasks;
+				
+				tasks = tasks.Where(t => t.Id == userId);
```

Comment:
Task.Id != Task.UserId

Reply:
In this situation I have a question. Which means the class below
 public class BaseEntity : IBaseEntity
    {
        public Guid Id { get; set; }
    }

Reply:
@KonstantinZh this is a base class for an entity that contains its id.
`UserId` of the `Completed` task is stored in its parent entity - `TaskEntity`.